### PR TITLE
Add a manual trigger option

### PR DESCRIPTION
**Pull request by mmitche**
Adds the ability to create a pipeline that is only manually triggered.  This isn't so much a trigger as just a way of creating the job.  Jobs are created on the trigger call, so this allows for a job that isn't triggered from outside of a manual action.

### CODE REVIEW

**Comment by mmitche** [comment]
@dotnet-bot test ci please

**Comment by mmitche** [comment]
@dotnet-bot test this please

**Comment by mmitche** [comment]
@dotnet/dnceng Can someone PTAL?

File: src/org/dotnet/ci/triggers/GenericTriggerBuilder.groovy [hunk 1]
```diff
@@ -27,16 +28,25 @@ class GenericTriggerBuilder implements TriggerBuilder {
...
+     * @param cronString Cron string to run the job on
+     * @return New periodic trigger that runs on the specified interval
+     */
     def static GenericTriggerBuilder triggerPeriodically(String cronString) {
         def newTrigger = new GenericTriggerBuilder(TriggerType.PERIODIC)
         newTrigger._cronString = cronString
         return newTrigger
     }
+
+    /* Constructs a new manual trigger.  This effecitvley means no trigger
```
Comment (maririos) [inline src/org/dotnet/ci/triggers/GenericTriggerBuilder.groovy]: Typo `effecitvley `

File: src/org/dotnet/ci/triggers/GenericTriggerBuilder.groovy [hunk 2]
```diff
@@ -7,7 +7,8 @@ import jobs.generation.Utilities
 // Periodic - Periodic triggers against a Git repo
```
Comment (maririos) [inline src/org/dotnet/ci/triggers/GenericTriggerBuilder.groovy]: To be consistent, add Manual description, or delete accordingly

File: src/org/dotnet/ci/triggers/GenericTriggerBuilder.groovy [hunk 3]
```diff
@@ -7,7 +7,8 @@ import jobs.generation.Utilities
...
 class GenericTriggerBuilder implements TriggerBuilder {
```
Comment (maririos) [inline src/org/dotnet/ci/triggers/GenericTriggerBuilder.groovy]: Out of curiosity, do you see other types of Triggers being created in the future? Will it be better to refactor this into an interface and make Periodic and Manual extend from `GenericTriggerBuilder`? Is that even possible in Groovy? :|

File: src/org/dotnet/ci/pipelines/Pipeline.groovy [hunk 4]
```diff
@@ -328,6 +328,19 @@ class Pipeline {
         return triggerPipelineOnEvent(builder, parameters)
     }
 
+    /* Creates a Triggers a pipeline that only triggers manually
```
Comment (chcosta) [inline src/org/dotnet/ci/pipelines/Pipeline.groovy]: nit: awkward grammar "Createa a Triggers a pipeline"

File: src/org/dotnet/ci/triggers/GenericTriggerBuilder.groovy [hunk 5]
```diff
@@ -48,19 +58,23 @@ class GenericTriggerBuilder implements TriggerBuilder {
     // Parameters:
     //  job - Job to emit trigger for
     void emitTrigger(def job) {
-        assert this._triggerType == TriggerType.PERIODIC
+        assert (this._triggerType == TriggerType.PERIODIC) || (this._triggerType == TriggerType.MANUAL)
```
Comment (chcosta) [inline src/org/dotnet/ci/triggers/GenericTriggerBuilder.groovy]: You could consider refactoring this into something like

```
if (this._triggerType == TriggerType.Periodic) {
 ...
}
else if (this._trigerType == TriggerType.Manual) {
 ...
}
else {
    assert(blah) /* Unsupported trigger type */
}
```

May simplify adding trigger types in the future.

**Comment by chcosta** [comment]
What's the real world scenario for this?

**Comment by mmitche** [comment]
@dotnet-bot test this please (testing CI)

File: src/org/dotnet/ci/triggers/GenericTriggerBuilder.groovy (hunk 3 above)
Comment (mmitche) [inline src/org/dotnet/ci/triggers/GenericTriggerBuilder.groovy]: Yeah, interfaces are possible (see TriggerBuilder.groovy).

I don't see other types coming in here.

**Comment by mmitche** [comment]
@chcosta For in-development jobs.  It disables the automatic triggers but still generates the job (otherwise trigger-less jobs don't get created)

**Comment by mmitche** [comment]
@dotnet-bot test this please

**Pull request merged**